### PR TITLE
feat: task 4

### PR DESCRIPTION
**Pull request by Zwezh**
Frontend application is integrated with products service (/products API) and products from product-service are represented on Frontend: https://dstzz2s2kvye2.cloudfront.net
Link to FE REPO PR - https://github.com/Zwezh/shop-angular-cloudfront/pull/3
SwaggerHub link - https://app.swaggerhub.com/apis/rs-products/product/1.0.0
**Task 4.1**: SQL scripts :
```
create table products (
	id uuid primary key default uuid_generate_v4(),
	title text not null,
	description text,
	price integer
)
create table stocks (
	product_id uuid,
	count integer,
	foreign key(product_id) references products(id)
)

insert into products (title, description, price) values
	('Nissan Leaf', 'The Nissan Leaf is a compact five-door hatchback battery electric vehicle (BEV) manufactured by Nissan. It was introduced in Japan and the United States in December 2010, and it is currently in its second generation, introduced in October 2017.', 12500),
	('Renault Zoe', 'The Renault Zoe is a five-door supermini electric car produced by the French manufacturer Renault. Renault originally unveiled, under the Zoe name, a number of different concept cars. Initially in 2005 as the Zoe City Car and later as the Zoe Z.E. electric concept was shown in two different versions in 2009 and 2010 under the Renault Z.E. name. A production ready version of the Zoe was shown at the 2012 Geneva Motor Show.', 22500),
	('BMW i3', 'The BMW i3 is a B-segment, high-roof hatchback manufactured and marketed by BMW with an electric powertrain using rear wheel drive via a single-speed transmission and an underfloor Li-ion battery pack and an optional range-extending petrol engine. The i3 was BMWs first mass-produced zero emissions vehicle and was launched as part of BMWs electric vehicle BMW i sub-brand.', 45000),
	('Tesla Model 3', 'The Model 3 is an electric fastback mid-size four-door sedan developed by Tesla.[7] The Model 3 Standard Range Plus version delivers an EPA-rated all-electric range of 263 miles (423 km) and the Long Range versions deliver 353 miles (568 km). According to Tesla, the Model 3 carries full self-driving hardware, with periodic software updates adding functionality.', 50000),
	('Jaguar I-Pace', 'The Jaguar I-Pace (stylised as "I-PACE") is a battery-electric crossover SUV produced by Jaguar Land Rover (JLR), the British subsidiary of the Indian automotive company Tata, under their Jaguar marque. The I-Pace was announced in March 2018, European deliveries began in June 2018, North American deliveries started in October 2018, and Indian deliveries began in March 2021.', 81000),
	('NIO ES8', 'The NIO ES8 is an all-electric, 7-seater full-sized Sport Utility Vehicle manufactured by NIO. The ES8 was put into production in June 2018 for the Chinese market.', 65000),
	('Tesla Model S', 'The Tesla Model S is an all-electric five-door liftback produced by Tesla, Inc. It was introduced on June 22, 2012.[12] It received a major refresh in June 2021. Tesla positions the Model S as its flagship.', 80000),
	('Audi e-tron', 'The Audi e-tron is an all-electric, midsize luxury[3] crossover produced by Audi, which was unveiled as a concept car at the 2015 Frankfurt Motor Show. It is the companys first electric mass production car, and was first delivered in May 2019. It has an EPA range of 204 miles (328 km), or 222 miles (357 km) in the updated model.', 100000),
	('Volkswagen ID.4', 'The ID.4 is one of the first all-electric compact SUVs to hit the market, offering a ton of standard features and a genuinely spacious cabin. It pushes the envelope with some neat technology and has a generous 250 miles of range. There is a lot to get excited about here.', 45000),
	('Peugeot e-208', 'On the list from 2020, the e-208 is a great little hatchback and one that we see as great value for the price. One of the most popular cars in the UK, the Peugeot 208 now comes with a fully electric 50kWh battery providing 138 horsepower.', 32000),
	('Vauxhall Corsa-e', 'The Vauxhall Corsa-e is very much a competitor to the Peugeot e-208. With a similar range of 209 miles and at a similar price range theres not much to choose from between these two. The Corsa-e comes with rapid charging capability allowing you to charge 80% in 30 minutes from a 100kW charging station.', 31550),
	('MINI Electric', 'Keeping many of the features that Mini fans know and love, with its signature style and its agile handling, it’s getting rave reviews. We think its introduction will be a bit of a game-changer to the electric car world.', 28000),
	('Porsche Taycan 4S','The Taycan represents Porsches foray into the electric vehicle space, and it does not disappoint. The Taycan is a technological masterpiece that manages to deliver incredible comfort and effortless driving one moment and outstanding performance the next.', 150000),
	('Ford Mustang Mach-E','The Mustang Mach-E is Fords first electric vehicle built from the ground up, and they knocked it out of the park. From an engaging driving experience to a comfortable interior brimming with technology, the Mustang Mach-E is simply the most well-rounded luxury EV on sale today.', 60000),
	('Hyundai Kona Electric','The Kona Electric is a new favorite among the more accessible EVs. It offers an impressive 258 miles of range, an attractive interior, and a long list of tech and safety features. The Kona is also surprisingly quick and entertaining to drive.', 40000),
	('Chevrolet Bolt EV','The Chevrolet Bolt boasts a 259-mile range, quick acceleration and an approachable price. Even with more competitors now, the Bolt is certainly worth checking out if you are in the market for an EV.', 40000),
	('Hyundai Ioniq Electric','The Hyundai Ioniq Electric is a tremendous value for an electric car, both in terms of cost of entry and efficiency. It also offers all the user-friendly tech we expect from Hyundai. But it falls short of class leaders in terms of performance and driving range at just 124 miles.', 36000)
insert into stocks (product_id, count) values
	('1ea51b34-ac6d-4700-b6a2-6732ca1f7238', 10),
	('00a4fef3-13eb-45d5-aec8-23db0e4b8911', 8),
	('6ce11f5e-f80e-4ea9-96a8-2ac18c005a21', 3),
	('346cf1dc-dd2f-4f1e-b17f-67c9ed310704', 10),
	('da137809-399e-47e5-af86-829b4d8a4e20', 20),
	('c610f242-1472-429e-8844-eb3e6f350b10', 14),
	('708d8f80-13d2-4c16-b484-8cd2705dc522', 15),
	('c1ae41fd-a745-4da6-9921-87ec06e61059', 18),
	('67401be2-1323-4b84-9b8e-1caf863c84eb', 22),
	('973f55ec-c7ff-4dc0-adcc-d9f63d890d2d', 35),
	('22128859-af22-41de-8565-f7722b633deb', 21),
	('3f1cde29-058a-4d63-a4c7-9bdf253b52ce', 77),
	('206d5883-420e-46a2-8d6e-9a9f5cbb42a3', 13),
	('576d6ff6-7e27-45bb-bd1e-6adafb022877', 16),
	('d32c0280-7f89-431f-9335-7697ddd591d8', 16),
	('66b341c5-d29e-44b9-9e4d-53662a5cd3e2', 18),
	('5b537766-f17b-4b72-ae0e-ab4f362a519f', 3)
create extension if not exists "uuid-ossp"
```
**Task 4.2:**
Get products - https://8lqt13zye6.execute-api.eu-west-1.amazonaws.com/products
Get product by id - https://8lqt13zye6.execute-api.eu-west-1.amazonaws.com/products/67401be2-1323-4b84-9b8e-1caf863c84eb
**Task 4.3:**
Create product/POST request - https://8lqt13zye6.execute-api.eu-west-1.amazonaws.com/products
```
{
    title: string - required,
    description: string - required,
    price: integer - required,
    count: integer - required
}
```

Additional (optional) tasks are done as well:
+1 (All languages) - POST/products lambda functions returns error 400 status code if product data is invalid
+1 (All languages) - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)

### CODE REVIEW

File: productCreate.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Client } from 'pg';
+import { dbOptions } from './db-options';
+
+export const handler = async (event) => {
+  const client = new Client(dbOptions);
```
Comment (AlinaLoz) [inline productCreate.js:5]: зачем открываь коннект к базе, если у тебя запрос с клиента будет неправильным?

**Comment by LemToUp** [comment]
7/9

I had doubt about "All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)"
because GET product by wrong Id returns 200 code.

File: productCreate.js (hunk 1 above)
Comment (Zwezh) [reply productCreate.js:5]: Все верно, по существо, исправлю

File: productCreate.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  await client.connect();
+  try {
+    const { title, description, price, count } = JSON.parse(event.body);
+    if (!title || !description || !price || !count)
```
Comment (AlinaLoz) [inline productCreate.js:9]: 1. что если строка продукта 1 символ. можно поставить милимальную и максимальную длину строки
2. price < 0
3. price 222222222222222222222222222222, c переполнением. такая же ситуация с count

Comment (Zwezh) [reply productCreate.js:9]: Спасибо. Будет исправлено.

File: productCreate.js (hunk 1 above)
Comment (Zwezh) [reply productCreate.js:12]: Исправлено

File: productCreate.js (hunk 2 above)
Comment (Zwezh) [reply productCreate.js]: Исправлено

**Pull request merged**